art/src/node.rs

```diff
@@ -476,6 +476,22 @@ impl<V> NodeBox<V> {
     pub fn is_null(&self) -> bool {
         self.inner == 0
     }
+
+    /// Unboxes `self` and returns its containing value.
+    ///
+    /// # Panics
+    ///
+    /// Panics if it does not contain `NodeBodyV`.
+    pub fn into_value(self) -> V {
+        let ptr = self.inner & !TAG_MASK;
+        assert_eq!(self.inner & TAG_MASK, 4);
```

Comment:
How about let it return `Option<V>` rather than asserting here?

Reply:
This code is designed to follow the convention of `NodeBox::into` -- it panics if `self` does not contain a kind of `NodeBodyI`. I agree that `Option<V>` is better but if so, APIs are a bit less consistent, IMO.